source/dub/commandline.d

```diff
@@ -383,16 +650,90 @@ class CommandArgs {
 		m_args = null;
 	}
 
-	/** Returns the list of unprocessed arguments and calls `dropAllArgs`.
+	/** Returns the list of unprocessed arguments, ignoring the app arguments,
+	and calls `dropAllArgs`.
 	*/
 	string[] extractRemainingArgs()
 	{
 		auto ret = m_args[1 .. $];
 		m_args = null;
 		return ret;
 	}
+
+	/** Returns the list of unprocessed arguments, including the app arguments
+		and calls `dropAllArgs`.
+
+	Returns:
+		Returns the list of unprocessed arguments with the app arguments
+	*/
+	string[] extractAllRemainingArgs()
+	{
+		auto ret = extractRemainingArgs();
```

Comment:
The "calling dropAllArgs" part here starts not making any sense anymore. Just looks like a trap someone could fall into because this function will return something different the second time it's called. In the current usage in this file this becomes extremely error prone because there are 2 different `extractAllRemainingArgs` functions (in here and in CommandNameArgument) which do completely different things.

Reply:
There is no `CommandNameArgument` struct anymore. Is this still a problem?

Reply:
hm I think it's ok now. However now because potentially you could more easily misuse the API (calling both extractRemainingArgs and extractAllRemainingArgs), can you add an assert into extractRemainingArgs to check if m_args has been extracted yet and otherwise error that you should only call one of the functions once?

Reply:
done